src/syntax/parser.mly

```diff
@@ -933,49 +934,63 @@ and parse_complex_type_inner = parser
 		| [< >] -> serror())
 	| [< '(Question,p1); t,p2 = parse_complex_type_inner >] ->
 		CTOptional (t,p2),punion p1 p2
+	| [< n = dollar_ident; s >] ->
+		(match s with parser
+		| [< '(DblDot,_); t = parse_complex_type_inner >] ->
+			let p1 = snd n in
+			let p2 = snd t in
+			CTNamed (n,t),punion p1 p2
+		| [< s >] ->
```

Comment:
Does it make a difference? I called it outside of the pattern because we need to unpack `n` to `n,p`, but we might as well just use `(fst n)`, `(snd n)`. I'm not sure what's better performance-wise.